src/commands/build/upload.rs

```diff
@@ -106,6 +107,23 @@ pub fn make_command(command: Command) -> Command {
                 .long("release-notes")
                 .help("The release notes to use for the upload.")
         )
+        .arg(
+            Arg::new("git_metadata")
+                .long("git-metadata")
+                .num_args(0..=1)
+                .default_missing_value("true")
+                .value_parser(clap::value_parser!(bool))
+                .help("Controls whether to collect and send git metadata (branch, commit, etc.). \
+                    Use --git-metadata to force enable, --git-metadata=false or --no-git-metadata to force disable. \
+                    If not specified, git metadata is automatically collected only when running in a CI environment.")
+        )
+        .arg(
+            Arg::new("no_git_metadata")
+                .long("no-git-metadata")
+                .action(ArgAction::SetTrue)
+                .conflicts_with("git_metadata")
+                .help("Disable collection of git metadata. Equivalent to --git-metadata=false.")
+        )
```

Comment:
To keep things simple, let's provide only a single way to disable Git metadata. I would be OK with either removing the `--no-git-metadata` flag, per the suggestion below, or changing the `--git-metadata` flag to only be a simple flag which does not take a true/false value.

```suggestion
```